sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/TableIdentifierParserSuite.scala

```diff
@@ -30,13 +64,22 @@ class TableIdentifierParserSuite extends SparkFunSuite {
     // Illegal names.
     intercept[ParseException](parseTableIdentifier(""))
     intercept[ParseException](parseTableIdentifier("d.q.g"))
+  }
 
+  test("table identifier - keywords") {
     // SQL Keywords.
-    val keywords = Seq("select", "from", "where", "left", "right")
+    val keywords = Seq("select", "from", "where") ++ hiveNonReservedRegression
     keywords.foreach { keyword =>
       intercept[ParseException](parseTableIdentifier(keyword))
       assert(TableIdentifier(keyword) === parseTableIdentifier(s"`$keyword`"))
       assert(TableIdentifier(keyword, Option("db")) === parseTableIdentifier(s"db.`$keyword`"))
     }
   }
+
+  test("table identifier - non reserved keywords") {
+    // Hive keywords are allowed.
+    hiveNonReservedKeyword.foreach { nonReserved =>
+      assert(TableIdentifier(nonReserved) === parseTableIdentifier(nonReserved))
+    }
+  }
```

Comment:
@gatorsmile I think the metastore will allow names starting with underscores, since Hive allows names starting with underscores if you put the name in backticks. However we are talking about Hive, and it has surprised me before. Could you open a PR to test if the metastore allows this?